Makefile

```diff
@@ -149,111 +110,54 @@ e2e_targets := test-e2e $(e2e_tests)
 .PHONY: test-e2e-setup
 export KIND_CLUSTER := operator-sdk-e2e
 export KUBECONFIG := $(HOME)/.kube/kind-$(KIND_CLUSTER).config
-export KUBEBUILDER_ASSETS := $(PWD)/tools/bin
+export KUBEBUILDER_ASSETS := $(PWD)/$(TOOLS_DIR)
 test-e2e-setup: build
-	./tools/scripts/fetch kind 0.9.0
-	./tools/scripts/fetch envtest 0.6.3
-	./tools/scripts/fetch kubectl ${K8S_VERSION} # Install kubectl AFTER envtest because envtest includes its own kubectl binary
-	[[ "`./tools/bin/kind get clusters`" =~ "$(KIND_CLUSTER)" ]] || ./tools/bin/kind create cluster --image="kindest/node:v$(K8S_VERSION)" --name $(KIND_CLUSTER)
+	$(SCRIPTS_DIR)/fetch kind 0.9.0
+	$(SCRIPTS_DIR)/fetch envtest 0.6.3
+	$(SCRIPTS_DIR)/fetch kubectl $(K8S_VERSION) # Install kubectl AFTER envtest because envtest includes its own kubectl binary
+ifeq (,$(findstring $(KIND_CLUSTER),$(shell $(TOOLS_DIR)/kind get clusters -q)))
+	$(TOOLS_DIR)/kind create cluster --image="kindest/node:v$(K8S_VERSION)" --name $(KIND_CLUSTER)
+endif
 
 .PHONY: test-e2e-teardown
 test-e2e-teardown:
-	./tools/scripts/fetch kind 0.9.0 && ./tools/bin/kind delete cluster --name $(KIND_CLUSTER)
+	$(SCRIPTS_DIR)/fetch kind 0.9.0
+	$(TOOLS_DIR)/kind delete cluster --name $(KIND_CLUSTER)
 	rm -f $(KUBECONFIG)
 
 # Double colon rules allow repeated rule declarations.
 # Repeated rules are exectured in the order they appear.
-$(e2e_targets):: test-e2e-setup image-build-scorecard-test
+$(e2e_targets):: test-e2e-setup image/scorecard-test
 
 test-e2e:: $(e2e_tests) ## Run e2e tests
-test-e2e-go:: image-build-custom-scorecard-tests ## Run Go e2e tests
+test-e2e-go:: image/custom-scorecard-tests ## Run Go e2e tests
 	go test ./test/e2e-go -v -ginkgo.v
-test-e2e-ansible:: image-build-ansible ## Run Ansible e2e tests
+test-e2e-ansible:: image/ansible-operator ## Run Ansible e2e tests
 	go test -count=1 ./internal/ansible/proxy/...
 	go test ./test/e2e-ansible -v -ginkgo.v
-test-e2e-ansible-molecule:: image-build-ansible ## Run molecule-based Ansible e2e tests
+test-e2e-ansible-molecule:: image/ansible-operator ## Run molecule-based Ansible e2e tests
 	./hack/tests/e2e-ansible-molecule.sh
-test-e2e-helm:: image-build-helm ## Run Helm e2e tests
+test-e2e-helm:: image/helm-operator ## Run Helm e2e tests
```

Comment:
I actually removed all the slashes from the `image` builds last year, in PRs #2019 & #2157. The slash method does reduce the duplication. I guess it's okay, I just find all the "reusable" targets hard to grep and figure out initially. I know about them now.

Reply:
Yeah the only reason I re-added `<type>/<pattern>` is for re-usability and de-cluttering, since Makefiles have great pattern matching/substitution functionality.